File: dotnet10-dapper/Dotnet10Dapper/Dto/FruitDto.cs
using System.ComponentModel.DataAnnotations;

namespace Dotnet10Dapper.Dto;

public record FruitDto(
    long? Id,
    [property: Required(ErrorMessage = "Name is mandatory")] string? Name,
    string? Description,
    IReadOnlyList<StoreFruitPriceDto>? StorePrices = null)
{
    public IReadOnlyList<StoreFruitPriceDto> StorePrices { get; init; } = StorePrices ?? [];
}

File: helidon4-se/src/main/java/org/acme/helidon/se/dto/FruitDTO.java
package org.acme.helidon.se.dto;

import java.util.List;

public record FruitDTO(Long id, String name, String description, List<StoreFruitPriceDTO> storePrices) {
    public FruitDTO {
        storePrices = storePrices == null ? List.of() : List.copyOf(storePrices);
    }
}
